group-coordinator/src/main/java/org/apache/kafka/coordinator/group/RecordHelpers.java

```diff
@@ -0,0 +1,283 @@
+                (short) 0
+            )
+        );
+    }
+
+    public static Record newGroupEpochTombstoneRecord(
+        String groupId
+    ) {
+        return new Record(
+            new ApiMessageAndVersion(
+                new ConsumerGroupMetadataKey()
+                    .setGroupId(groupId),
+                (short) 3
+            ),
+            null // Tombstone.
+        );
+    }
+
+    public static Record newTargetAssignmentRecord(
+        String groupId,
+        String memberId,
+        Map<Uuid, Set<Integer>> partitions
+    ) {
+        return new Record(
+            new ApiMessageAndVersion(
+                new ConsumerGroupTargetAssignmentMemberKey()
+                    .setGroupId(groupId)
+                    .setMemberId(memberId),
+                (short) 7
+            ),
+            new ApiMessageAndVersion(
+                new ConsumerGroupTargetAssignmentMemberValue()
+                    .setTopicPartitions(partitions.entrySet().stream()
+                        .map(keyValue -> new ConsumerGroupTargetAssignmentMemberValue.TopicPartition()
+                            .setTopicId(keyValue.getKey())
+                            .setPartitions(new ArrayList<>(keyValue.getValue())))
+                        .collect(Collectors.toList())),
+                (short) 0
+            )
+        );
+    }
+
+    public static Record newTargetAssignmentTombstoneRecord(
+        String groupId,
+        String memberId
+    ) {
+        return new Record(
+            new ApiMessageAndVersion(
+                new ConsumerGroupTargetAssignmentMemberKey()
+                    .setGroupId(groupId)
+                    .setMemberId(memberId),
+                (short) 7
+            ),
+            null // Tombstone.
+        );
+    }
+
+    public static Record newTargetAssignmentEpochRecord(
+        String groupId,
+        int groupEpoch
+    ) {
+        return new Record(
+            new ApiMessageAndVersion(
+                new ConsumerGroupTargetAssignmentMetadataKey()
+                    .setGroupId(groupId),
+                (short) 6
+            ),
+            new ApiMessageAndVersion(
+                new ConsumerGroupTargetAssignmentMetadataValue()
+                    .setAssignmentEpoch(groupEpoch),
+                (short) 0
+            )
+        );
+    }
+
+    public static Record newTargetAssignmentEpochTombstoneRecord(
+        String groupId
+    ) {
+        return new Record(
+            new ApiMessageAndVersion(
+                new ConsumerGroupTargetAssignmentMetadataKey()
+                    .setGroupId(groupId),
+                (short) 6
+            ),
+            null // Tombstone.
+        );
+    }
+
+    public static Record newCurrentAssignmentRecord(
+        String groupId,
+        ConsumerGroupMember member
+    ) {
+        return new Record(
+            new ApiMessageAndVersion(
+                new ConsumerGroupCurrentMemberAssignmentKey()
```

Comment:
I see... The reason for putting `Member` and `Metadata` second in the target assignment case is because `Member` and `Metadata` are attributes of the target assignment. For the current assignment, we don't have this so I went with `ConsumerGroupCurrentMemberAssignmentKey/Value` which is a bit more readable in my opinion.

That being said, I am not attached to it at all. Should we just use `ConsumerGroupCurrentAssignmentKey/Value` without `Member`?